src/decimal.rs

```diff
@@ -1694,8 +1694,68 @@ const fn flags(neg: bool, scale: u32) -> u32 {
     (scale << SCALE_SHIFT) | ((neg as u32) << SIGN_SHIFT)
 }
 
+macro_rules! impl_try_from_decimal {
+    ($TInto:ty, $conversion_fn:path, $name:expr) => {
+        ///
+        /// Try to convert a `Decimal` to a primitive type.
+        ///
+        /// Can fail if the `Decimal` is out of range.
+        /// For integer types, truncate and return the integer part.
+        ///
+        impl core::convert::TryFrom<Decimal> for $TInto {
+            type Error = crate::Error;
+
+            #[inline]
+            fn try_from(t: Decimal) -> Result<Self, Error> {
+                $conversion_fn(&t).ok_or_else(|| Error::from(format!("Failed to convert to {}", $name)))
```

Comment:
nit: you could use `stringify!($TInto)` instead of $name here.

Reply:
Also, I know the precedent right now is to return a `String` for an `Error`, however it'd be preferable to introduce an new `enum` option within the `Error` type. Ideally, I'd like to move away from `ErrorString`.